### PR TITLE
unique envoy cluster ids

### PR DESCRIPTION
**Pull request by wasaga**
## Summary

- make sure envoy cluster names are unique
- keep cluster metrics name w/o prefix for now, subject to change after https://github.com/pomerium/pomerium-console/issues/645 

## Related issues

Fixes https://github.com/pomerium/internal/issues/290 

## Checklist

- [x] reference any related issues
- [ ] updated docs
- [x] updated unit tests
- [ ] updated UPGRADING.md
- [x] add appropriate tag (`improvement` / `bug` / etc)
- [x] ready for review

### CODE REVIEW

File: internal/controlplane/xds_cluster_test.go [hunk 1]
```diff
@@ -542,6 +542,25 @@ func Test_buildCluster(t *testing.T) {
 	})
 }
 
+func Test_validateClusters(t *testing.T) {
+	type c []*envoy_config_cluster_v3.Cluster
+	testCases := []struct {
+		clusters    c
+		expectError bool
+	}{
+		{c{{Name: "one"}, {Name: "one"}}, true},
```
Comment (desimone) [inline internal/controlplane/xds_cluster_test.go:551]: Should we test the error case?

File: internal/controlplane/xds_clusters.go [hunk 2]
```diff
@@ -407,3 +415,22 @@ func (srv *Server) buildTransportSocketMatches(endpoints []Endpoint) ([]*envoy_c
...
+func validateClusters(clusters []*envoy_config_cluster_v3.Cluster) error {
+	return validateClusterNamesUnique(clusters)
+}
+
+// validateClusterNamesUnique checks cluster names are unique, as they're effectively IDs
+func validateClusterNamesUnique(clusters []*envoy_config_cluster_v3.Cluster) error {
+	names := make(map[string]bool, len(clusters))
+
+	for _, c := range clusters {
+		if _, there := names[c.Name]; there {
```
Comment (desimone) [inline internal/controlplane/xds_clusters.go:429]: nit: `there` is a little confusing. `exists`?

File: internal/controlplane/xds_routes.go [hunk 3]
```diff
@@ -231,13 +231,23 @@ func (srv *Server) buildControlPlanePrefixRoute(prefix string, protected bool) (
 	return r, nil
 }
 
-var getPolicyName = func(policy *config.Policy) string {
-	if policy.EnvoyOpts != nil && policy.EnvoyOpts.Name != "" {
-		return policy.EnvoyOpts.Name
+// getClusterID returns cluster ID
```
Comment (desimone) [inline internal/controlplane/xds_routes.go:234]:
```suggestion
// getClusterID returns a cluster ID
```

**Review by calebdoxsey — APPROVED** [review]

File: internal/controlplane/xds_cluster_test.go (hunk 1 above)
Comment (wasaga) [reply internal/controlplane/xds_cluster_test.go:551]: this one is an error case (duplicate cluster names)

Comment (desimone) [reply internal/controlplane/xds_cluster_test.go:551]: Sorry, I mean the inverse. Basically to test both cases. Perhaps some test cases with different cases, and whitespace?

**Comment by desimone** [comment]
@wasaga fyi backport failed :/

**Pull request merged**